docs/developer-guide/working-with-custom-formatters.md

```diff
@@ -56,30 +80,25 @@ The output of the previous command will be something like this
         fixableErrorCount: 0,
         fixableWarningCount: 0
     }
-];
+]
 ```
 
-As you can see the argument passed to the custom formatter function is just a list of results objects.
-
-## Description of the results
-
-### the result object
+### The `result` Object
 
-You will receive a result object from each file `eslint` validates, each one of them containing
-the list of messages for `errors` and/or `warnings`.
+Each object in the `results` array is a `result` object. Each `result` object contains the path of the file that was linted and information about linting issues that were encountered. Here are the properties available on each `result` object:
 
-The following are the fields of the result object:
-
-*   **filePath**: The path to the file relative to the current working directory (the path from which ESLint was executed).
-*   **messages**: An array of message objects. See below for more info about messages.
+*   **filePath**: The absolute path to the file that was linted.
+*   **messages**: An array of `message` objects. See below for more info about messages.
 *   **errorCount**: The number of errors for the given file.
 *   **warningCount**: The number of warnings for the given file.
 *   **source**: The source code for the given file. This property is omitted if this file has no errors/warnings or if the `output` property is present.
 *   **output**: The source code for the given file with as many fixes applied as possible. This property is omitted if no fix is available.
```

Comment:
I think the documentation for these properties is duplicated from the [`CLIEngine` docs](https://eslint.org/docs/developer-guide/nodejs-api#cliengineexecuteonfiles). It seems like the duplication is probably worthwhile here to avoid too much redirection, but could we add an HTML comment near both of these sections to mention the duplication, to avoid one copy getting out of date when the other copy changes? For example:

```markdown
<!-- This section is copied from the "Node.js API" page. Changes to this section should
also be manually applied to that page. -->
```

Reply:
Thinking longer-term, I'd like to get documentation for these types of objects into the Node.js API docs as well (after some reorganization and after taking significant care to ensure readability and navigability). I consider these contracts to be part of the public API as well for semver purposes, and it would be nice if we could (at some point) say, "our public API is documented completely on the Node.js API web page".